### DISCUSSION TITLE
How to define and implement a custom nushell script with main which has multiple subcommands?

### DISCUSSION THREAD

**Discussion by avkonst** [nushell/nushell#6826] [Q&A]
Let's say I would like to implement an application which has multiple subcommands and flags specific to each subcommand, like docker for example. How can I do this? The docs do not give such an example :(

**Comment by fdncred** [comment]
Here's how you make commands and sub commands.

command
```
def some [] { print some }
```
sub command
```
def 'some param' [] { print param }
```

**Reply by avkonst** [reply]
and if my command is main, will it work too?

**Reply by fdncred** [reply]
Yes, I'm pretty sure I saw someone doing this last week. Either in an issue, a discussion, or on Discord.
